### ISSUE TITLE
Window border not present after restoring from maximised fullscreen

### ISSUE THREAD

**Issue by adamdb5** [microsoft/terminal#10607]
### Windows Terminal version (or Windows build number)

1.8.1521.0

### Other Software

Also effects Command Prompt and Powershell at the time of writing this (Windows 10.0.22000.0).

### Steps to reproduce

1. Open terminal
2. Maximise terminal window (Win+Up)
3. Enter fullscreen (F11)
4. Restore terminal window (Win+Down)

### Expected Behavior

The terminal window should restore with a border.

Although it could be argued that the window should return to its maximised state, as this behaviour is used by windows explorer.

<img width="892" alt="Capture1" src="https://user-images.githubusercontent.com/34384005/125160342-43824980-e174-11eb-8d3c-abb29eac5ba1.PNG">

### Actual Behavior

The terminal window restores to its previous location and size, however it does not have a border.

<img width="903" alt="Capture" src="https://user-images.githubusercontent.com/34384005/125160366-5ac13700-e174-11eb-9b73-f41ab7efbc88.PNG">

**Comment by zadjii-msft** [comment]
Holy crap why does this repro with conhost 😨 That's really really unexpected. I'd imagine that <kbd>win+down</kbd> shouldn't work at all from _fullscreen_, or if it does, it should at least toggle off the "fullscreen" window styles.

Out of curiosity, do you know if this repro'd in Windows 10? I wouldn't be surprised if this was a Win11 regression, but I'd be equally unsurprised if this was just always how the console worked.

**Comment by adamdb5** [comment]
I first saw the issue a few years ago in Windows 10 with cmd, and initially reported it via the feedback hub, but I don't think it ever got picked up.

Out of curiosity I re-enabled my insider account to test the issue on Windows 11 and it was still present. 

Then, I tested it on terminal, and it was also effected.

**Comment by zadjii-msft** [comment]
> initially reported it via the feedback hub, but I don't think it ever got picked up.

Yea, feedback hub is a bit of a black hole sometimes. At least if it did repro in Windows 10, then I don't need to go chase down the DWM team to figure out how it regressed. Now we just need to fix our code 😅

**Comment by adamdb5** [comment]
I attempted to find a solution myself, but was unsuccessful.

From what I found, `win+down` invokes `onResize` however `onResize` doesn't know the previous fullscreen state.

This means that if we were to check the value of `_fullscreen` in `onResize`, it would be `true` when entering fullscreen as well as when `win+down` is pressed (as we never *actually* exit fullscreen), so we cannot use the value of `_fullscreen` as a test.

I thought maybe `onRestore` may be invoked but that is only when transitioning from a minimized state. 

This is my first time looking at the terminal source, so I'm probably looking in the wrong places :smile:. I look forward to seeing the solution :smile:.
